packages/netsuite-adapter/src/filters/analytics_definition_handle.ts

```diff
@@ -91,9 +91,91 @@ const fetchTransformFunc = async (
+  instance: InstanceElement,
+  workbookToolsRecord: Record<string, Values>,
+  name: string
+): void => {
+  if (Array.isArray(instance.value[name])) {
+    instance.value[name].forEach((tool:unknown) => {
+      if (values.isPlainObject(tool)) {
+        Object.values(tool)
+          .filter(values.isPlainObject)
+          .forEach((obj: Values) => {
+            if (_.isString(obj[SOAP_SCRIPT_ID])) {
+              workbookToolsRecord[obj[SOAP_SCRIPT_ID]] = obj
+            }
+          })
+      }
+    })
+  }
+}
+
+const createRecordOfPivotsChartsAndDsLinks = async (
+  elementsSource: ReadOnlyElementsSource,
+  elemID: ElemID,
+): Promise<Record<string, Values>> => {
+  const oldWorkbookToolsRecord: Record<string, Values> = {}
+  const oldInstance = await elementsSource.get(elemID)
+  if (isInstanceElement(oldInstance)) {
+    addWorkbookToolsToRecord(oldInstance, oldWorkbookToolsRecord, PIVOTS)
+    addWorkbookToolsToRecord(oldInstance, oldWorkbookToolsRecord, CHARTS)
+    addWorkbookToolsToRecord(oldInstance, oldWorkbookToolsRecord, DATASET_LINKS)
+  }
+  return oldWorkbookToolsRecord
+}
+
+const isSameXmlValues = (xml1: string, xml2: string): boolean => {
+  const values1 = parse(xml1, {
+    attributeNamePrefix: ATTRIBUTE_PREFIX,
+    ignoreAttributes: false,
+    tagValueProcessor: val => decode(val),
+  })
+  const values2 = parse(xml2, {
+    attributeNamePrefix: ATTRIBUTE_PREFIX,
+    ignoreAttributes: false,
+    tagValueProcessor: val => decode(val),
+  })
+  return _.isEqual(values1, values2)
+}
+
+const adjustWorkbookToolsOrder = (workbookTools: Value[], oldWorkbookToolsRecord: Record<string, Values>): void => {
+  workbookTools
+    .filter(values.isPlainObject)
+    .flatMap(obj => Object.values(obj))
+    .forEach((workbookTool: Values) => {
+      if (_.isArray(workbookTool.datasets)) {
+        workbookTool.datasets.sort()
+      }
+      if (_.isString(workbookTool[SOAP_SCRIPT_ID]) && workbookTool[SOAP_SCRIPT_ID] in oldWorkbookToolsRecord) {
+        const oldPivot = oldWorkbookToolsRecord[workbookTool[SOAP_SCRIPT_ID]]
+        const innerXmlTitles = [DEFINITION, MAPPING]
+        innerXmlTitles.forEach((xmlTitle: string) => {
+          if (_.isString(workbookTool[xmlTitle]) && _.isString(oldPivot[xmlTitle])
+            && isSameXmlValues(workbookTool[xmlTitle], oldPivot[xmlTitle])) {
+            workbookTool[xmlTitle] = oldPivot[xmlTitle]
+          }
+        })
+      }
+    })
+}
+
+const discardUnrelevantChanges = (
+  newInstanceValues: Values,
+  oldTools: Record<string, Values>,
+): void => {
+  const innerArrayNames = [PIVOTS, CHARTS, DATASET_LINKS]
+  innerArrayNames.forEach((arrName: string) => {
+    if (Array.isArray(newInstanceValues[arrName])) {
+      adjustWorkbookToolsOrder(newInstanceValues[arrName], oldTools)
+    }
+  })
+}
+
 const createAnalyticsInstance = async (
   instance: InstanceElement,
   analyticsType: ObjectType,
+  elementsSource: ReadOnlyElementsSource | undefined,
```

Comment:
nit - `elementsSource`is always defined in filters
```suggestion
  elementsSource: ReadOnlyElementsSource,
```